lib/check-tags.js

```diff
@@ -0,0 +1,41 @@
+var log = require('./out')({});
+
+function checkTags(app, mod, name) {
+  if (mod.tags && mod.tags !== '' && typeof mod.tags === 'string') {
+    mod.tags = mod.tags.split(',');
+  }
+
+  function checkMatch (A, B) {
+    return A.filter(function(tag) {
+      return (B.indexOf(tag) !== -1);
+    });
+  }
+
+  function checkMatches (A, B) {
+    var result = checkMatch(A, B);
+    if (result.length !== 0) {
+      return result;
+    } else {
+      return false;
+    }
+  }
+
+  if ((app.withTags && mod.tags && !checkMatches(mod.tags, app.withTags)) ||
+      (app.withTags && !mod.tags)) {
```

Comment:
This logic seems unnecessarily complex, and comments would be good, see my previous suggestion.